src/topp/HighResPrecursorMassCorrector.cpp

```diff
@@ -275,6 +281,84 @@ class TOPPHiResPrecursorMassCorrector :
       return corrected_precursors;
     }
 
+    set<Size> correctToHighestIntensityMS1Peak(PeakMap & exp, double mz_tolerance, bool ppm, vector<double> & deltaMZs,vector<double> & mzs, vector<double> & rts)
+    {
+      set<Size> corrected_precursors;
+      // load experiment and extract precursors
+      vector<Precursor> precursors;  // precursor
+      vector<double> precursors_rt;  // RT of precursor MS2 spectrum
+      vector<Size> precursor_scan_index;
+      getPrecursors_(exp, precursors, precursors_rt, precursor_scan_index);
+
+      for (Size i = 0; i != precursors_rt.size(); ++i)
+      {
+        // get precursor rt
+        double rt = precursors_rt[i];
+
+        // get precursor MZ
+        double mz = precursors[i].getMZ();
+
+        // cout << rt << " " << mz << endl;
+
+        // get precursor spectrum
```

Comment:
the code is correct but the commend should be adapted:
the line retrieves an iterator on the MS2 fragment spectrum, the precursor spectrum is the MS1 you walk in line 310